scio-redis/src/main/scala/com/spotify/scio/redis/RedisCommand.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.redis
+
+import org.apache.beam.sdk.io.redis.{RedisIO => BeamRedisIO}
+
+/** Represents an abstract Redis command. */
+sealed trait RedisCommand extends Serializable
+
+/**
+ * A Redis command which mutates a Redis data structure.
+ * @tparam K Type of a key. Could be string or byte array.
+ * @tparam V Type of a value being transferred to Redis as a part of the mutation.
+ *           This might be a byte array, string, etc.
+ */
+final case class MutationRedisCommand[K, V](command: BeamRedisIO.Write.Method, key: K, value: V)
+    extends RedisCommand
+
+sealed trait RedisValueTypeEvidence[T]
+
+object RedisValueTypeEvidence {
+  implicit val stringValueType: RedisValueTypeEvidence[String] =
+    new RedisValueTypeEvidence[String] {}
+
+  implicit val byteArrayValueType: RedisValueTypeEvidence[Array[Byte]] =
+    new RedisValueTypeEvidence[Array[Byte]] {}
+}
+
+/**
+ * TODO: this isn't been used yet, need to change Beam RedisIO to support both binary and string
+ * keys and values.
+ */
+object RedisCommand {
```

Comment:
I think I see where you are going with this but AFAICT right now Beam `RedisIO` only allows `KV<String, String>` which is limited. To fully support dymanic commands we will need to have our own implementation of `RedisIO` as we have `ElasticsearchIO` due to the same reasons.

Our `RedisIO` would have a `RedisWriter[T <: RedisMutation]` where:

```scala
sealed abstract class RedisType[T]

object RedisType {
  implicit case object StringValueType extends RedisType[String]
  implicit case object ByteArrayValueType extends RedisType[Array[Byte]]
}

sealed abstract class RedisMutation

object RedisMutation {
  case class Append[T: RedisType](key: T, value: T) extends RedisMutation
  case class Set[T: RedisType](key: T, value: T) extends RedisMutation
  case class IncrBy[T: RedisType](key: T, value: Long) extends RedisMutation
  case class DecrBy[T: RedisType](key: T, value: Long) extends RedisMutation
  case class RPush[T: RedisType](key: T, value: T) extends RedisMutation

  ...
}
```

I suggest we use Beam's impl as a first iteration and follow up with our own impl. wdyt?

Reply:
Agree. Let's start from using this out-of-the-box IO supporting only strings. I am not going to include this change into this PR.